client/sites/tools/logs/hooks/use-actions.tsx

```diff
@@ -0,0 +1,90 @@
+import { copy } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import { LogType, ServerLog, PHPLog } from 'calypso/data/hosting/use-site-logs-query';
+import { useDispatch } from 'calypso/state';
+import { successNotice, errorNotice } from 'calypso/state/notices/actions';
+import DetailsModalPHP from '../components/details-modal-php';
+import DetailsModalServer from '../components/details-modal-server';
+
+const useActions = ( { logType, isLoading }: { logType: LogType; isLoading: boolean } ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const actions = useMemo( () => {
+		if ( logType === LogType.PHP ) {
+			return [
+				{
+					id: 'copy-msg',
+					label: translate( 'Copy message' ),
+					icon: copy,
+					isPrimary: true,
+					disabled: isLoading,
+					supportsBulk: false,
+					callback: async ( items: ( PHPLog | ServerLog )[] ) => {
+						const message = ( items[ 0 ] as PHPLog ).message;
+						try {
+							await navigator.clipboard.writeText( message );
+							dispatch(
+								successNotice(
+									/* translators: notice shown upon copy of Logs entry */
+									translate( 'Copied' )
```

Comment:
I was testing these. and a few things stood up for potential improvement:

* Can we be more clear about what we're copying? "Copied" seems a bit too short. I'd go with "Copied message to clipboard" or "Copied request URL to clipboard" similar to our messages in core.
* For PHP logs, we're copying the message. Is that enough? Could it be more helpful to copy the entire entry that includes the timestamp?
* For web server logs, we're copying just the request URL, I'd suggest being able to copy more of the entry.

Reply:
I've address the copy suggestions at https://github.com/Automattic/wp-calypso/pull/100827